lib/scraper/manual_input_sheet.rb

```diff
@@ -0,0 +1,34 @@
+module Scraper
+  class ManualInputSheet < ScraperBase
+    def scrape
+      manual_input_sheet = ManualInputSheet.new
+      manual_input_sheet.load_sheet
+      advocacy_campaigns = manual_input_sheet.advocacy_campaigns
+      create_campaigns(advocacy_campaigns)
+    end
+
+    private
+
+    def create_campaigns(advocacy_campaigns)
+      advocacy_campaigns.each do |campaign|
+        create_campaign(campaign)
+      end
+    end
+
+    def create_campaign(campaign)
+      find_or_create_campaign(campaign)
+    rescue Exception => e
+      # Rescuing all exceptions is typically a terrible idea.
+      # We're doing it here because we always want to ensure the scraper can
+      # continue iterating through the list of scraped campaigns.
+
+      log_scrape_failure(e, campaign_data)
+    end
```

Comment:
Looks a lot like https://github.com/RagtagOpen/cta-scraper/blob/29866e0f0bf8325a71cf12a07283c1fb96055758/lib/scraper/five_calls.rb#L37-L45

What do you think about moving this method up to the ScraperBase class and eliminating the duplication?

Reply:
I’m good with that! I originally didn’t do that to make sure I didn’t paint myself into a corner, but it does look like that would be ok to do.